x-pack/plugins/stack_alerts/server/alert_types/es_query/lib/fetch_search_source_query.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { buildRangeFilter, Filter } from '@kbn/es-query';
+import { Logger } from 'kibana/server';
+import { OnlySearchSourceAlertParams } from '../types';
+import {
+  getTime,
+  ISearchSource,
+  ISearchStartSearchSource,
+  SortDirection,
+} from '../../../../../../../src/plugins/data/common';
+
+export async function fetchSearchSourceQuery(
+  alertId: string,
+  params: OnlySearchSourceAlertParams,
+  latestTimestamp: string | undefined,
+  services: {
+    logger: Logger;
+    searchSourceClient: Promise<ISearchStartSearchSource>;
+  }
+) {
+  const { logger, searchSourceClient } = services;
+  const client = await searchSourceClient;
+  const initialSearchSource = await client.create(params.searchConfiguration);
+
+  const { searchSource, dateStart, dateEnd } = updateSearchSource(
+    initialSearchSource,
+    params,
+    latestTimestamp
+  );
+
+  logger.debug(
+    `search source query alert (${alertId}) query: ${JSON.stringify(
+      searchSource.getSearchRequestBody()
+    )}`
+  );
+
+  const searchResult = await searchSource.fetch();
```

Comment:
Yes for sure these searches can last long, and we can cancel them , could you provide some details how to do it? we could solve it in a follow up for sure. This could also be the case when using the default search type, so I think it should also be applied there.

Reply:
Opened an issue for this #128043